From 52460b29f4d8ca1e545f0250546d4087aa293b32 Mon Sep 17 00:00:00 2001
From: Anders Wallin <anders.wallin@windriver.com>
Date: Sat, 30 Dec 2017 17:09:40 +0100
Subject: [PATCH] tmpfix: using nroff instead of man

Replaced 'man' with 'nroff' to support VERY old build machines that does not
support 'man' argument
--no-hyphenation --no-justification

Signed-off-by: Anders Wallin <anders.wallin@windriver.com>
---
 configure.ac        | 6 +++---
 doc/man/Makefile.am | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 38308ac..8eb7434 100644
--- a/configure.ac
+++ b/configure.ac
@@ -742,9 +742,9 @@ AS_IF([test "x$embedded_help" = "xyes"], [
 	AS_IF([test "x$man_pages_opt" = "xno"], [
 		AC_MSG_ERROR([You need the --enable-man-pages option with the --enable-embedded-help option.])
 	])
-	AC_PATH_PROG([man_prog_path], [man], [no])
-	AS_IF([test "x$man_prog_path" = "xno"], [
-		AC_MSG_ERROR([You need man with the --enable-embedded-help option.])
+	AC_PATH_PROG([nroff_prog_path], [nroff], [no])
+	AS_IF([test "x$nroff_prog_path" = "xno"], [
+		AC_MSG_ERROR([You need nroff with the --enable-embedded-help option.])
 	])
 	AC_DEFINE_UNQUOTED([LTTNG_EMBED_HELP], 1, [Embed --help messages.])
 	AC_SUBST([MANPROG], [$man_prog_path])
diff --git a/doc/man/Makefile.am b/doc/man/Makefile.am
index eb88e86..a696601 100644
--- a/doc/man/Makefile.am
+++ b/doc/man/Makefile.am
@@ -82,7 +82,7 @@ MAN8_H = $(addsuffix .8.h,$(MAN8_NAMES))
 MAN_H = $(MAN1_H) $(MAN3_H) $(MAN8_H)
 MAN_H_RECIPE = \
 	$(mantoh_verbose_0)\
-	MANWIDTH=80 @MANPROG@ --encoding=UTF-8 --no-hyphenation --no-justification --local-file $< > $@ ; \
+	nroff -t -Tutf8 -mandoc $< | col -bpx > $@ ; \
 	$(SED) -i 's/\\/\\\\/g' $@ ; \
 	$(SED) -i 's/"/\\"/g' $@ ; \
 	$(SED) -i 's/^\(.*\)$$/"\1\\n"/' $@
-- 
1.8.3.1

